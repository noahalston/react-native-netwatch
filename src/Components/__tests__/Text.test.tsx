import * as React from 'react';
import { StyleSheet } from 'react-native';
import { shallow } from 'enzyme';
import { Text, Title, TextSecondaryColor, IProps } from '../Text';

describe('Status test suite', () => {
  let props: IProps;

  it('should render Text properly', () => {
    givenProps('Children text', StyleSheet.create({ text: { color: 'red' } }));
    expect(shallow(<Text {...props} />)).toMatchSnapshot();
  });

  it('should render Title properly', () => {
    givenProps('Children text', StyleSheet.create({ text: { color: 'red' } }));
    expect(shallow(<Title {...props} />)).toMatchSnapshot();
  });

  it('should render Title properly', () => {
    givenProps('Children text', StyleSheet.create({ text: { color: 'red' } }));
    expect(shallow(<TextSecondaryColor {...props} />)).toMatchSnapshot();
  });

  function givenProps(children: string, style: any) {
    props = {
      children,
      style,
    };
  }
});
