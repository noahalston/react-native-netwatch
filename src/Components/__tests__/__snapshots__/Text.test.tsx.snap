// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status test suite should render Text properly 1`] = `
<ForwardRef(withTheme(undefined))
  style={
    Object {
      "text": Object {
        "color": "red",
      },
    }
  }
>
  Children text
</ForwardRef(withTheme(undefined))>
`;

exports[`Status test suite should render Title properly 1`] = `
<ForwardRef(withTheme(undefined))
  style={
    Object {
      "text": Object {
        "color": "red",
      },
    }
  }
>
  Children text
</ForwardRef(withTheme(undefined))>
`;

exports[`Status test suite should render Title properly 2`] = `
<ForwardRef(withTheme(undefined))
  style={
    Object {
      "text": Object {
        "color": "red",
      },
    }
  }
>
  Children text
</ForwardRef(withTheme(undefined))>
`;
